src/algorithm/nest_cfgs.rs

```diff
@@ -624,7 +624,7 @@ pub(crate) mod test {
         Ok((split, merge))
     }
 
-    fn build_then_else_merge_from_if<T: HugrMutRef>(
```

Comment:
do these actually need both Mut and Ref or is the Mut requirement sufficient?

Reply:
They need it because the `impl` block is also parameterized like that...as I say I think we could probably go through and replace quite a few `AsRef + AsMut`s with one or the other, but I've not done that thorough an analysis here; consistency is helpful in itself and we can reconsider later if we really want.